src/mbgl/style/parser.cpp

```diff
@@ -116,6 +116,74 @@ StyleParseResult Parser::parse(const std::string& json) {
         }
     }
 
+#ifdef MLN_TEXT_SHAPING_HARFBUZZ
+    // Ignore font-faces if no harfbuzz
+    if (document.HasMember("font-faces")) {
+        const JSValue& faces = document["font-faces"];
+        if (faces.IsObject()) {
+            fontFaces = std::make_shared<FontFaces>();
+            for (auto it = faces.MemberBegin(); it != faces.MemberEnd(); ++it) {
+                const std::string& faceName = it->name.GetString();
+                const JSValue& faceValue = it->value;
+
+                if (faceValue.IsArray()) {
+                    // If the face is an array, we assume it is a list of font file objects.
+                    for (const auto& fontFile : faceValue.GetArray()) {
+                        if (fontFile.IsObject()) {
+                            FontFace fontFace;
+                            fontFace.name = faceName;
+
+                            // Parse url
+                            if (fontFile.HasMember("url")) {
+                                const JSValue& url = fontFile["url"];
+                                if (url.IsString()) fontFace.url = url.GetString();
+                            }
+
+                            // Parse unicode-range
+                            if (fontFile.HasMember("unicode-range")) {
+                                const JSValue& unicodeRange = fontFile["unicode-range"];
+                                if (unicodeRange.IsArray()) {
+                                    for (auto& range : unicodeRange.GetArray()) {
+                                        if (range.IsString()) {
+                                            std::string rangeString = range.GetString();
+                                            if (rangeString.length() > 2) {
+                                                rangeString = rangeString.substr(2);
+                                                std::string::size_type pos = rangeString.find('-');
+                                                if (pos != std::string::npos) {
+                                                    std::string start = rangeString.substr(0, pos);
+                                                    std::string end = rangeString.substr(pos + 1);
+                                                    if (!start.empty() && !end.empty()) {
+                                                        int startInt = std::stoi(start, nullptr, 16);
```

Comment:
This will throw an exception in case of an invalid number. I am not sure if we are handling that well on each platform.